internal/app/auth-srv.go

```diff
@@ -0,0 +1,211 @@
+package app
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/caos/oidc/pkg/op"
+	"github.com/gorilla/mux"
+	"github.com/labstack/echo/v4"
+	"github.com/reearth/reearth-backend/internal/app/appauth"
+	"github.com/reearth/reearth-backend/internal/usecase/interactor"
+	"github.com/reearth/reearth-backend/internal/usecase/interfaces"
+
+	"github.com/golang/gddo/httputil/header"
+)
+
+func authEndPoints(ctx context.Context, e *echo.Echo, r *echo.Group, cfg *ServerConfig) {
+
+	userUsecase := interactor.NewUser(cfg.Repos, cfg.Gateways, cfg.Config.SignupSecret)
+
+	domain, err := url.Parse(cfg.Config.Auth.Domain)
+	if err != nil {
+		panic("not valid auth domain")
+	}
+	domain.Path = "/"
+
+	config := &op.Config{
+		Issuer:                domain.String(),
+		CryptoKey:             sha256.Sum256([]byte(cfg.Config.Auth.Key)),
+		GrantTypeRefreshToken: true,
+	}
+	storage := appauth.NewAuthStorage(&appauth.StorageConfig{
+		Domain: domain.String(),
+		Debug:  cfg.Debug,
+	})
+	handler, err := op.NewOpenIDProvider(
+		ctx,
+		config,
+		storage,
+		op.WithHttpInterceptors(jsonToFormHandler()),
+		op.WithHttpInterceptors(setURLVarsHandler()),
+		op.WithCustomKeysEndpoint(op.NewEndpoint(".well-known/jwks.json")),
+	)
+	if err != nil {
+		e.Logger.Fatal(err)
+	}
+
+	router := handler.HttpHandler().(*mux.Router)
+
+	err = router.Walk(muxToEchoMapper(r))
+	if err != nil {
```

Comment:
```suggestion
	if err := router.Walk(muxToEchoMapper(r)); err != nil {
```

Minimize variable scopes